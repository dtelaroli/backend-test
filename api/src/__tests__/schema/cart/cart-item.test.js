const { PRODUCT } = require("../../../db/seeders/consts");
const {
  queries: { GET_CART },
  mutations: { ADD_CART_ITEM, UPDATE_CART_ITEM, DELETE_CART_ITEM, CREATE_SKU },
} = require("../graphql");
const { seed, mutate, query } = require("../../helpers");

describe("CartItem", () => {
  beforeEach(() => {
    seed();
  });

  describe("Mutation", () => {
    it("Should create cart, add, update and remove a item to cart", async () => {
      const get = await query({
        mutation: GET_CART,
        variables: {
          id: "321",
        },
      });

      const createSku = await mutate({
        mutation: CREATE_SKU,
        variables: {
          input: {
            id: "321",
            sku: "S123",
            inventory: 10,
            price: 9.99,
            productId: PRODUCT.id,
          },
        },
      });

      const cart = get.data.getCart;
      const sku = createSku.data.createSku;

      const add = await mutate({
        mutation: ADD_CART_ITEM,
        variables: {
          input: {
            cartId: cart.id,
            skuId: sku.id,
            quantity: 10,
          },
        },
      });

      const update = await mutate({
        mutation: UPDATE_CART_ITEM,
        variables: {
          input: {
            cartId: cart.id,
            skuId: sku.id,
            quantity: 1,
          },
        },
      });

      const remove = await mutate({
        mutation: DELETE_CART_ITEM,
        variables: {
          input: {
            cartId: cart.id,
            skuId: sku.id,
          },
        },
      });

      expect({ get, add, update, remove }).toMatchSnapshot();
    });
  });
});
