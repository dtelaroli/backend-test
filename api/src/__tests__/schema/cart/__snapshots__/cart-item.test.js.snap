// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Mutation Should create cart, add, update and remove a item to cart 1`] = `
Object {
  "add": Object {
    "data": Object {
      "addCartItem": Object {
        "id": "321",
        "items": Array [
          Object {
            "cartId": "321",
            "itemAmount": 99.9,
            "itemQuantity": 10,
            "sku": Object {
              "id": "321",
              "inventory": 10,
              "price": 9.99,
              "sku": "S123",
            },
            "skuId": "321",
          },
        ],
        "totalAmount": 99.9,
        "totalQuantity": 10,
      },
    },
    "errors": undefined,
    "extensions": undefined,
    "http": Object {
      "headers": Headers {
        Symbol(map): Object {
          "Cache-Control": Array [
            "max-age=5, public",
          ],
        },
      },
    },
  },
  "get": Object {
    "data": Object {
      "getCart": Object {
        "id": "321",
        "items": Array [],
        "totalAmount": 0,
        "totalQuantity": 0,
      },
    },
    "errors": undefined,
    "extensions": undefined,
    "http": Object {
      "headers": Headers {
        Symbol(map): Object {
          "Cache-Control": Array [
            "max-age=5, public",
          ],
        },
      },
    },
  },
  "remove": Object {
    "data": Object {
      "deleteCartItem": Object {
        "id": "321",
        "items": Array [],
        "totalAmount": 0,
        "totalQuantity": 0,
      },
    },
    "errors": undefined,
    "extensions": undefined,
    "http": Object {
      "headers": Headers {
        Symbol(map): Object {
          "Cache-Control": Array [
            "max-age=5, public",
          ],
        },
      },
    },
  },
  "update": Object {
    "data": Object {
      "updateCartItem": Object {
        "id": "321",
        "items": Array [
          Object {
            "cartId": "321",
            "itemAmount": 9.99,
            "itemQuantity": 1,
            "sku": Object {
              "id": "321",
              "inventory": 10,
              "price": 9.99,
              "sku": "S123",
            },
            "skuId": "321",
          },
        ],
        "totalAmount": 9.99,
        "totalQuantity": 1,
      },
    },
    "errors": undefined,
    "extensions": undefined,
    "http": Object {
      "headers": Headers {
        Symbol(map): Object {
          "Cache-Control": Array [
            "max-age=5, public",
          ],
        },
      },
    },
  },
}
`;
